src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -165,8 +165,14 @@ public AzureAppConfigurationOptions Select(string keyFilter, string labelFilter
             {
                 throw new ArgumentException("The characters '*' and ',' are not supported in label filters.", nameof(labelFilter));
             }
-
-            if (!_kvSelectors.Any(s => string.Equals(s.KeyFilter, keyFilter) && string.Equals(s.LabelFilter, labelFilter)))
+            var foundExactSameSelector = _kvSelectors.Find(s => string.Equals(s.KeyFilter, keyFilter) && string.Equals(s.LabelFilter, labelFilter));
```

Comment:
Since the type is not very obvious here, we should use explicit type instead of `var` for readability.